### PR TITLE
Expose middleware composition util

### LINKED ISSUE #58: Expose applyConfigMiddleware util
https://github.com/LaunchPadLab/lp-requests/blob/master/src/http/apply-config-middleware.js

### PR DESCRIPTION
**Pull request by dpikt**
Resolves #58 

This will help us on projects with `before` hooks that are getting unwieldy.

### CODE REVIEW

File: src/http/compose-middleware.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+ */
+
+function composeMiddleware (...middlewares) {
+  // Note: An async reducer is used because we want to allow middleware
+  // to return promises if they want.
+  const asyncMiddlewares = middlewares.map(ensureAsync)
+  return function callMiddleware (initialValue) {
+    return asyncReduce(
+      asyncMiddlewares,
+      (acc, middleware) => middleware(acc).then(result => merge(acc, result || {})),
```
Comment (chawes13) [inline src/http/compose-middleware.js]: Is it worth trying to safeguard against keys with an undefined value here? That probably breaks Single Responsibility, but would help if users forget to add guard clauses to each individual middleware

Comment (dpikt) [reply src/http/compose-middleware.js]: I don't think so. The way we've used undefined keys before has been a work-around to handle the use case that guard clauses provide- if someone returns a key with an undefined value from middleware, I'd assume they want it to be passed in.

Comment (dpikt) [reply src/http/compose-middleware.js]: LMK if that idea makes sense or if you'd like to appeal 😄

Comment (chawes13) [reply src/http/compose-middleware.js]: 👍 Yep! That's where I was leaning but wanted to check. Let's not bastardize it just for my use case 😏

**Pull request merged**